### PR TITLE
Sort custom prompt list by recency

### PR DESCRIPTION
**Pull request by Emt-lin**
see: #638

### CODE REVIEW

File: src/customPromptProcessor.ts [hunk 1]
```diff
@@ -52,7 +65,13 @@ export class CustomPromptProcessor {
         content: content,
       });
     }
-    return prompts;
+
+    // Clean up promptUsageTimestamps
+    this.usageStrategy?.removeUnusedPrompts(prompts.map((prompt) => prompt.title)).save();
+
+    console.log(this.settings.promptUsageTimestamps);
```
Comment (logancyang) [inline src/customPromptProcessor.ts:72]: We can remove this.

File: src/PromptUsageStrategy.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import { CopilotSettings } from "@/settings/SettingsPage";
```
Comment (logancyang) [inline src/PromptUsageStrategy.ts:1]: Nit: For consistency can we name this file `promptUsageStrategy.ts` without the capitalization.

**Comment by logancyang** [comment]
@Emt-lin Thank you so much for the PR! Generally lgtm, just some nits.

Btw, you can use `git rebase master` in the future, no need to merge and create a new commit.

**Comment by logancyang** [comment]
Merging now and will address the nits myself in the next PR.

**Comment by Emt-lin** [comment]
> @Emt-lin Thank you so much for the PR! Generally lgtm, just some nits.
> 
> Btw, you can use `git rebase master` in the future, no need to merge and create a new commit.

ok, Thank you for the reminder.

**Pull request merged**